packages/components/src/modal/use-modal-exit-animation.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * WordPress dependencies
+ */
+import { useReducedMotion } from '@wordpress/compose';
+import { useCallback, useRef, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { CONFIG } from '../utils';
+
+// Animation duration (ms) extracted to JS in order to be used on a setTimeout.
+const FRAME_ANIMATION_DURATION = CONFIG.transitionDuration;
+const FRAME_ANIMATION_DURATION_NUMBER = Number.parseInt(
+	CONFIG.transitionDuration
+);
+
+const EXIT_ANIMATION_NAME = 'components-modal__disappear-animation';
+
+export function useModalExitAnimation() {
+	const frameRef = useRef< HTMLDivElement >();
+	const [ isAnimatingOut, setIsAnimatingOut ] = useState( false );
+	const isReducedMotion = useReducedMotion();
+
+	const closeModal = useCallback(
+		( onAnimationEnd: () => void ) => {
```

Comment:
A little idea: what if the `closeModal` function returned a promise? Then it could be used as:
```js
onClick = () => closeModal().then( () => onRequestClose() );
```
Looks quite elegant to me.

Reply:
Love it! I switched to returning a promise in `closeModal` in https://github.com/WordPress/gutenberg/pull/65203/commits/0ab921dadfde8e40c7a69714b1ded43d88f2c701